lib/core/task.toit

```diff
@@ -75,6 +76,90 @@ interface Task:
   static current -> Task:
     return Task_.current
 
+  /**
+  Divides the work given by $lambdas up into a number of
+    separate tasks that run concurrently.
+
+  Returns the results of running the $lambdas as a $Map
+    keyed by the lambda index. If the lambda at a given
+    index did not return, the results map will not contain
+    and entry for it. The results map is insertion ordered,
+    so it is possible to tell which lambda returned first.
+
+  If any of $lambdas throw an exception, the exception is
+    propagated to the caller of $Task.divide which in
+    returns also throws.
+
+  If $required is less than the number of $lambdas, the
+    method returns when $required tasks have completed.
+  */
+  static divide lambdas/List -> Map
+      --required/int=lambdas.size:
+    count ::= lambdas.size
+    tasks ::= Array_ count
+    results ::= {:}
+    if count < 2 or not (1 <= required <= count):
+      throw "Bad Argument"
```

Comment:
We usually use "INVALID_ARGUMENT", but we need to go through all 'throw' places anyway, so I don't mind too much.
```suggestion
      throw "INVALID_ARGUMENT"
```

Reply:
Maybe I picked the wrong place for inspiration :)